src/macrofier/engine.mjs

```diff
@@ -1140,46 +1214,47 @@ function insertExampleMacros(template, examples, method, json, templates) {
   return template.replace(/\$\{method\.examples\}/g, content)
 }
 
-function generateResult(result, json, templates) {
-  const type = types.getSchemaType(result, json, { destination: state.destination, section: state.section  })
+function generateResult(result, json, templates, { name = '' } = {}) {
+
+  const type = types.getSchemaType(result, json, { name: name, destination: state.destination, section: state.section  })
 
   if (result.type === 'object' && result.properties) {
     let content = getTemplate('/types/object', templates).split('\n')
 
     for (var i=0; i<content.length; i++) {
       if (content[i].indexOf("${property}") >= 0) {
-        content[i] = Object.entries(result.properties).map(([title, property]) => insertSchemaMacros(content[i], title, property, json)).join('\n')
+        content[i] = Object.entries(result.properties).map(([title, property]) => insertSchemaMacros(content[i], title || name, property, json)).join('\n')
       }
     }
 
-    return insertSchemaMacros(content.join('\n'), result.title, result, json)
+    return insertSchemaMacros(content.join('\n'), name, result, json)
   }
   else if (type === 'string' && Array.isArray(result.enum)) {
-    return insertSchemaMacros(getTemplate('/types/enum', templates), result, json)
+    return insertSchemaMacros(getTemplate('/types/enum', templates), name, result, json)
   }
   else if (result.$ref) {
-    const link = getLinkForSchema(result, json)
+    const link = getLinkForSchema(result, json, { name: name})
 
     // if we get a real link use it
     if (link !== '#') {
       return `[${types.getSchemaType(result, json, { destination: state.destination, section: state.section  })}](${link})`
     }
     // otherwise this was a schema with no title, and we'll just copy it here
     else {
-      const schema = localizeDependencies(result, json)
+      const sch = localizeDependencies(result, json)
```

Comment:
Any reason for this shorter name?

Reply:
with previous changes, the variable schema was already used, hence tried a shorter name. But now that logic was changed and now only one schema variable so will change it back from sch to schema